src/app/file-manager/breadcrumb/breadcrumb.component.ts

```diff
@@ -23,30 +28,43 @@ export class BreadcrumbComponent implements OnInit {
 
   }
 
-  stepClicked(evt, segment) {
-    this.change.emit(segment);
+  stepClicked(evt, segment, index) {
+    this.change.emit(
+      {
+        index: index,
+        item: this.items[index]
+      }
+    );
   }
 
   homeClicked(evt) {
     this.home.emit('home');
   }
 
-  private processPath(fullpath: string) {
-    if (fullpath === null || fullpath === undefined) { return; }
-    fullpath = fullpath.substring(fullpath.length - 1, fullpath.length) === this.DELIMITER
-               ? fullpath.substr(0, fullpath.lastIndexOf(this.DELIMITER)) : fullpath;
-    const segs = fullpath.split(this.DELIMITER);
+  private processItems(steps: Item[]) {
     this.segments = [];
-    let p = '';
-    segs.forEach(s => {
-      p = p + s;
-      const e = {
-        label: s,
-        path: p
-      };
-      if (e.path !== '') {
+
+    if (steps === null || steps === undefined || steps.length <= 0) {
+      return;
+    }
+
+    steps.forEach(
+      (item, index) => {
+        const paths = item.prefix.split('/');
```

Comment:
The code defines `DELIMITER` above but it's not used at all in this class. Is this a case where it should be used?